Chinese Academy of Sciences CNIC CSTR.js

```diff
@@ -0,0 +1,194 @@
+	}
+	Z.debug(url);
+	ZU.doGet(url, parseJSON);
+}
+
+function detectWeb(doc, url) {
+	var searchRe = /^https?:\/\/(?:([^.]+\.))?(?:cstr\.cn)\/(?:search\/cstrDetail\/).*/;
+	if (searchRe.test(url)) {
+		return "journalArticle";	
+	}
+	else {
+		return "multiple";
+	}
+}
+// 解析页面元素获取元数据项目
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == 'multiple') {
+		var rows = doc.querySelectorAll('a.identifier');
+		if(rows){
+			var urls = [];
+			for (var i = 0; i < rows.length; i++) {
+				var id = rows[i].href.replace(/^https?:\/\/(?:([^.]+\.))?(?:cstr\.cn)\//,"");
+				urls.push('https://www.cstr.cn/openapi/v2/pid-cstr-service/detail?identifier=' + encodeURIComponent(id)
+				);
+			}
+			if(urls && urls.length > 0){
+				ZU.doGet(urls, parseJSON);
+			}
+		}		
+	}
+	else {
+		var id = doc.querySelector('a.identifier');		
+		if (id) {
+			id = id.innerText.replace(/^CSTR:\s*|v\d+|\s+.*$/ig, '');
+			var apiurl = 'https://www.cstr.cn/openapi/v2/pid-cstr-service/detail?identifier=' + encodeURIComponent(id);
+			ZU.doGet(apiurl, parseJSON);
+		}else{
+			if (!id) throw new Error('Could not find CSTR ID on page.');
+		}
+	}
+}
+
+function parseJSON(text) {
+	var obj = JSON.parse(text);
+	Z.debug(obj.code);
+
+	var code = obj.code;
+	if(code == 200){
+		var newItem = new Zotero.Item("journalArticle");
+		var content = obj.data.content;
+		newItem.title = content.resourceChineseName;
+		var identifier = content.identifier;
+		var creators = content.creators;
+		if (creators && creators.length) {
+			// arXiv.org format
+			for(var i =0 ; i<creators.length; i++){
+				// var creator_name = ZU.xpathText(creator, './/creator_name_cn').trim();
+				newItem.creators.push(
+					ZU.cleanAuthor(creators[i].creatorNameCN, "author", true)
+				);
+			}
+			
+		}
+		newItem.date = content.publicationDate;
+		var description = content.descriptionCN;
+		newItem.abstractNote = ZU.trimInternal(description);
+		newItem.notes.push({ note: identifier });
+		
+		newItem.extra = identifier;
+		newItem.url = "https://cstr.cn/"+identifier;
+		newItem.publicationTitle = identifier+ " " + content.submitOrgName;
+
+		var subjects = content.subjectClassifications;
+		for (let j = 0; j < subjects.length; j++) {
+			var subject = subjects.keyWordsCN;
+			newItem.tags.push(subject);
+		}
+	}	
+
+	// retrieve and supplement publication data for published articles via DOI
+	if (newItem.DOI) {
+		var translate = Zotero.loadTranslator("search");
+		// CrossRef
+		translate.setTranslator("b28d0d42-8549-4c6d-83fc-8382874a5cb9");
+		
+		var item = { itemType: "journalArticle", DOI: newItem.DOI };
+		translate.setSearch(item);
+		translate.setHandler("itemDone", function (obj, item) {
+			// Z.debug(item)
+			newItem.volume = item.volume;
+			newItem.issue = item.issue;
+			newItem.pages = item.pages;
+			newItem.date = item.date;
+			newItem.ISSN = item.ISSN;

```

Comment:
Should we be checking for existing values for these fields? Or will DOI always be as good / better?